flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/lsm/compact/IntervalPartition.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.storage.filestore.lsm.compact;
+
+import org.apache.flink.table.data.binary.BinaryRowData;
+import org.apache.flink.table.runtime.generated.RecordComparator;
+import org.apache.flink.table.storage.filestore.lsm.SortedRun;
+import org.apache.flink.table.storage.filestore.lsm.sst.SstFileMeta;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.stream.Collectors;
+
+/** Algorithm to partition several sst files into the minimum number of {@link SortedRun}s. */
+public class IntervalPartition {
```

Comment:
Why not the input is sst files?
So there is another class to produce `SortedRun`?

Reply:
`SortedRun` ensures that the files are sorted. We can, of course, accept a list of files and sort them by ourselves but in an lsm for all levels >= 1 the files are already sorted and I think we can take advantage of this.

Let's change the input for now and come back to this if we need this optimization.